db-tests/src/test/java/gov/nasa/jpl/aerie/database/ActivityDirectiveChangelogTests.java

```diff
@@ -0,0 +1,266 @@
+            start_offset = '01:01:01',
+            arguments = '{"biteSize": 2}'
+          WHERE plan_id = %d and id = %d
+          RETURNING *;
+          """.formatted(planId, activityId));
+    }
+    final var actDirMid = getActivity(planId, activityId);
+
+    revertActivityDirectiveToChangelog(planId, activityId, 0);
+    final var actDirAfter = getActivity(planId, activityId);
+
+    assertNotEquals(actDirBefore.name(), actDirMid.name());
+    assertNotEquals(actDirBefore.startOffset(), actDirMid.startOffset());
+    assertNotEquals(actDirBefore.arguments(), actDirMid.arguments());
+
+    assertEquals(actDirBefore.name(), actDirAfter.name());
+    assertEquals(actDirBefore.startOffset(), actDirAfter.startOffset());
+    assertEquals(actDirBefore.arguments(), actDirAfter.arguments());
+  }
+
+  private Activity getActivity(final int planId, final int activityId) throws SQLException {
+    try (final var statement = connection.createStatement()) {
+      final var res = statement.executeQuery(
+          // language=sql
+          """
+          SELECT *
+          FROM activity_directive
+          WHERE id = %d
+          AND plan_id = %d;
+          """.formatted(activityId, planId));
+
+      res.next();
+
+      return new Activity(
+          res.getInt("id"),
+          res.getInt("plan_id"),
+          res.getString("name"),
+          res.getInt("source_scheduling_goal_id"),
+          res.getString("created_at"),
+          res.getString("last_modified_at"),
+          res.getString("last_modified_by"),
+          res.getString("start_offset"),
+          res.getString("type"),
+          res.getString("arguments"),
+          res.getString("last_modified_arguments_at"),
+          res.getString("metadata"),
+          res.getString("anchor_id"),
+          res.getBoolean("anchored_to_start")
+      );
+    }
+  }
+
+  private void revertActivityDirectiveToChangelog(int planId, int activityDirectiveId, int revision) throws SQLException {
+    try (final var statement = connection.createStatement()) {
+      statement.executeQuery(
+          // language=sql
+          """
+          SELECT hasura_functions.restore_activity_from_changelog(
+            _plan_id => %d,
+            _activity_directive_id => %d,
+            _revision => %d,
+            hasura_session => '%s'::json);
+          """.formatted(planId, activityDirectiveId, revision, merlinHelper.admin.session())
+      );
+    }
+  }
+  private int getChangelogRevisionCount(int planId, int activityDirectiveId) throws SQLException {
+    try (final var statement = connection.createStatement()) {
+      final var res = statement.executeQuery(
+          // language=sql
+          """
+          SELECT count(revision)
+          FROM activity_directive_changelog
+          WHERE plan_id = %d and activity_directive_id = %d;
+          """.formatted(planId, activityDirectiveId)
+      );
+      res.next();
+      return res.getInt(1);
+    }
```

Comment:
Two things:

1) Please move helper functions to be between the `Before` and `After` functions and the start of the tests, surrounded by the comment markers `//region Helper Functions` `//endregion`to match the style in the other DB test files
2) Since the `getActivities` function appears to be copied from PlanCollaborationTests, it and the `Activity` record both ought to be pulled out into `MerlinDBHelper`. If doing so would require significant refactors to `PlanCollaborationTests`, then this can be left for a later PR

Reply:
1. I was following the style of `MerlinDatabaseTests.java`, although it looks like that style is somewhat outdated.
2. I figured DB test refactoring could be a separate unit of work, since there is also `Activity` record overlap in `AnchorTests` and `PresetTests`

Reply:
1. Yeah, MerlinDatabaseTests needs a refactor.
2. Fair. Made #1099 to cover the refactor.